src/world.rs

```diff
@@ -130,19 +131,19 @@ impl World {
             std::process::exit(0);
         }
 
-
         if is_mouse_button_down(macroquad::input::MouseButton::Left) {
+            self.mouse_down = true;
+        } else if self.mouse_down {
             let mouse_position: Vec2 = macroquad::input::mouse_position().into();
             let primary_ball = self.primary_ball_mut();
-            let d = primary_ball.position().distance(mouse_position);
-            let dx = primary_ball.position().x - mouse_position.x;
-            let dy = primary_ball.position().y - mouse_position.y;
-
-            let vx = (dx / d).sin();
-            let vy = (dy / d).sin();
-            let v = Vec2::new(vx, vy);
-            dbg!(v);
-            primary_ball.push(-v * 2.0);
+            let d = primary_ball.position().distance(mouse_position).max(0.001);
+            let v = (*primary_ball.position() - mouse_position) / d;
+            let screen_size: f32 =
+                f64::from(screen_width().powi(2) + screen_height().powi(2)).sqrt() as f32;
+            let limit: f32 = 30.0;
```

Comment:
this should be in some constant

Reply:
v and screen_size cannot be constant.
Only limit could be declared as a constant, but it won't be obvious what it refers to

Reply:
I was referring to `limit`